### PR TITLE
fix: Importing files using a resource query

### PR DESCRIPTION
**Pull request by dimitarnestorov**
https://github.com/peerigon/extract-loader/issues/53

### CODE REVIEW

**Comment by smac89** [comment]
Please merge this. Very useful

File: src/extractLoader.js [hunk 1]
```diff
@@ -91,7 +91,8 @@ function evalDependencyGraph({loaderContext, src, filename, publicPath = ""}) {
             exports,
             __webpack_public_path__: publicPath, // eslint-disable-line camelcase
             require: givenRelativePath => {
-                const indexOfQuery = givenRelativePath.includes("?") ? givenRelativePath.indexOf("?") : givenRelativePath.length;
+                const noDoubleQuestionMarkPath = givenRelativePath.replace(/\?\?/g, "##");
```
Comment (smac89) [inline src/extractLoader.js]: When is `??` part of a valid url? Also if you handled `??`, why didn't you also handle `???`?

I don't think you need to handle these anyways. You should just focus on fixing the query part which was handled wrongly in the original code and leave these other cases to the judgement of the person using it.

File: src/extractLoader.js [hunk 2]
```diff
@@ -91,7 +91,8 @@ function evalDependencyGraph({loaderContext, src, filename, publicPath = ""}) {
...
+                const indexOfQuery = noDoubleQuestionMarkPath.includes("?") ? noDoubleQuestionMarkPath.indexOf("?") : givenRelativePath.length;
```
Comment (smac89) [inline src/extractLoader.js]: A shorter way of accomplishing the same thing is:

`const indexOfQuery = givenRelativePath.search(/(\?.*)?$/);`

File: src/extractLoader.js (hunk 1 above)
Comment (dimitarnestorov) [reply src/extractLoader.js]: There was a test case failing, it was related to double question mark. I believe it is valid in webpack context.

File: src/extractLoader.js (hunk 2 above)
Comment (dimitarnestorov) [reply src/extractLoader.js]: I love it but code is meant to be readable and understandable. I'm going to leave the decision to the maintainer.

**Comment by jannikkeye** [comment]
Fixed in #66.

**Pull request closed without merging**